indexing-service/src/main/java/io/druid/indexing/common/task/Task.java

```diff
@@ -128,6 +135,20 @@
   public boolean isReady(TaskActionClient taskActionClient) throws Exception;
 
   /**
+   * Returns whether or not this task can restore its progress from its on-disk working directory. Restorable tasks
+   * may be started with a non-empty working directory. Tasks that exit uncleanly may still have a chance to attempt
+   * restores, meaning that restorable tasks should be able to deal with potentially partially written on-disk state.
+   */
+  public boolean canRestore();
```

Comment:
That could be done with EBS/EFS or NFS or some other type of strategy, yes.

Also imagine a world where some cluster resource manager like Mesos is holding a persistent resource for the runner so it can restart if needed. So there needs to be a clear expectation about how long Mesos (or the framework on Mesos) is expected to reserve that persistent resource before considering it lost.